common-rest/src/main/java/org/uniprot/api/rest/validation/ValidDownloadByIdsRequest.java

```diff
@@ -0,0 +1,152 @@
+package org.uniprot.api.rest.validation;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+import java.util.regex.Pattern;
+
+import javax.validation.Constraint;
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+import javax.validation.Payload;
+
+import lombok.extern.slf4j.Slf4j;
+
+import org.apache.commons.beanutils.BeanUtils;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.web.context.support.SpringBeanAutowiringSupport;
+import org.uniprot.core.util.Utils;
+import org.uniprot.store.config.UniProtDataType;
+import org.uniprot.store.config.returnfield.config.ReturnFieldConfig;
+import org.uniprot.store.config.returnfield.factory.ReturnFieldConfigFactory;
+
+/**
+ * @author sahmad
+ * @created 26/07/2021
+ */
+@Constraint(validatedBy = ValidDownloadByIdsRequest.AccessionListValidator.class)
+@Target({ElementType.TYPE, ElementType.ANNOTATION_TYPE})
+@Retention(RetentionPolicy.RUNTIME)
+public @interface ValidDownloadByIdsRequest {
```

Comment:
Maybe we should Rename it to use post in the name, instead of download.

Reply:
Agree